framework/test_app/runners/k8s/k8s_base_runner.py

```diff
@@ -193,8 +193,13 @@ def stop_pod_dependencies(self, *, log_drain_sec: int = 0):
         self.maybe_stop_logging()
 
         # Stop port forwarders if any.
-        for pod_port_forwarder in self.pod_port_forwarders:
-            pod_port_forwarder.close()
+        if self.pod_port_forwarders:
+            for pod_port_forwarder in self.pod_port_forwarders:
```

Comment:
[Minor] I don't think we need an `if` here, `for..in..` should be able to handle empty list.

Reply:
Yea - I agree. This is purely for the log statement below. I used to have it just for the log statement, but then logically it made more sense to group both for loop and logging under the same if.

Reply:
Fixed